keras_cv/layers/spatial_pyramid.py

```diff
@@ -142,21 +144,28 @@ def call(self, inputs, training=None):
         """Calls the Atrous Spatial Pyramid Pooling layer on an input.
 
         Args:
-          inputs: A `tf.Tensor` of shape [batch, height, width, channels]
+          inputs: A tensor of shape [batch, height, width, channels]
 
         Returns:
-          A `tf.Tensor` of shape [batch, height, width, num_channels]
+          A tensor of shape [batch, height, width, num_channels]
         """
         result = []
 
         for channel in self.aspp_parallel_channels:
-            temp = tf.cast(channel(inputs, training=training), inputs.dtype)
+            temp = ops.cast(channel(inputs, training=training), inputs.dtype)
             result.append(temp)
 
-        result = tf.concat(result, axis=-1)
+        result[-1] = keras.layers.Resizing(
+            inputs.shape[1], inputs.shape[2], interpolation="bilinear"
+        )(result[-1])
+
+        result = ops.concatenate(result, axis=-1)
         result = self.projection(result, training=training)
         return result
 
+    def compute_output_shape(self, input_shape):
```

Comment:
Why is this method needed in Keras Core?

Reply:
It's needed for tracing the shape while tracing the forward pass. Without it we get an error like this one:
```
Could not automatically infer the output shape / dtype of 'spatial_pyramid_pooling_9' (of type SpatialPyramidPooling). 
Either the `SpatialPyramidPooling.call()` method is incorrect, or you need to implement the `SpatialPyramidPooling.compute_output_spec() / compute_output_shape()` method. Error encountered:
```